addons/crm_iap_lead/models/crm_iap_lead_mining_request.py

```diff
@@ -0,0 +1,237 @@
+        _logger.info('Start Lead Mining Requests Processing')
+        requests = self.search([('state', '=', 'pending')])
+        for req in requests:
+            server_payload = self._prepare_iap_payload(req)
+            enough_credit = self._perform_request(req, server_payload)
+            if autocommit:
+                # auto-commit for batch processing
+                self._cr.commit()
+            if not enough_credit:
+                break       # stops the processing of this request and the following requests
+        _logger.info('End Lead Mining Requests Processing')
+
+    @api.model
+    def _prepare_iap_payload(self, req):
+        """
+        This will prepare the data to send to the server
+        """
+        payload = {'lead_number': req.lead_number,
+                   'search_type': req.search_type,
+                   'countries': req.country_ids.mapped('code')}
+        if req.state_ids:
+            payload['states'] = req.state_ids.mapped('code')
+        if req.filter_on_size:
+            payload.update({'company_size_min': req.company_size_min,
+                            'company_size_max': req.company_size_max})
+        if req.technology_ids:
+            payload['technology_tags'] = req.technology_ids.mapped('tech_tag')
+        if req.search_type == 'people':
+            payload.update({'contact_number': req.contact_number,
+                            'contact_filter_type': req.contact_filter_type})
+            if req.contact_filter_type == 'role':
+                payload.update({'preferred_role': req.preferred_role_id.reveal_id,
+                                'other_roles': req.role_ids.mapped('reveal_id')})
+            elif req.contact_filter_type == 'seniority':
+                payload['seniority'] = req.seniority_id.reveal_id
+        return payload
+
+    @api.model
+    def _perform_request(self, req, server_payload):
+        """
+        This will perform the request and create the corresponding leads.
+        The user will be notified if he hasn't enough credits.
+        """
+        reveal_account = self.env['iap.account'].get('reveal')
+        dbuuid = self.env['ir.config_parameter'].sudo().get_param('database.uuid')
+        endpoint = self.env['ir.config_parameter'].sudo().get_param('reveal.endpoint', DEFAULT_ENDPOINT) + '/iap/clearbit/1/lead_mining_request'
+        params = {
+            'account_token': reveal_account.account_token,
+            'dbuuid': dbuuid,
+            'data': server_payload
+        }
+        response = jsonrpc(endpoint, params=params, timeout=300)
+        if response.get('credit_error'):
+            self.env['crm.iap.lead.helpers'].notify_no_more_credit('lead_mining', self._name, 'lead.mining.already_notified')
+            return False
+        else:
+            self.env['ir.config_parameter'].sudo().set_param('lead.mining.already_notified', False)
+            self._create_leads_from_response(req, response['data'])
```

Comment:
remove rq parameter from function, return response['data'], and self._create_leads_from_response and set state done in 'action_submit' if the response is not False